Assets/Scripts/Enemies/Enemy.cs

```diff
@@ -147,80 +145,79 @@ public class Enemy : MonoBehaviour {
             return false;
         }
     }
-  
+
     public void ApplyStatusEffect(StatusEffect statusEffect) {
         activeStatusEffects.Add(statusEffect);
-        
+
         RefreshStatusEffects();
     }
-    
+
     public void RemoveStatusEffect(StatusEffect statusEffect) {
         activeStatusEffects.Remove(statusEffect);
-        
+
         RefreshStatusEffects();
     }
-    
-    public void ApplyTimedStatusEffect(StatusEffect statusEffect, Tower originTower)
-    {
+
+    public void ApplyTimedStatusEffect(StatusEffect statusEffect, Tower originTower) {
         TimedStatusEffect newTimedStatusEffect = new TimedStatusEffect(statusEffect, originTower);
-        
+
         bool statusEffectMissing = true;
-        
+
         foreach (TimedStatusEffect timedStatusEffect in timedStatusEffects) {
             if (newTimedStatusEffect.Equals(timedStatusEffect)) {
                 timedStatusEffect.RefreshTimer();
                 statusEffectMissing = false;
             }
         }
-        
+
         if (statusEffectMissing) {
             timedStatusEffects.Add(newTimedStatusEffect);
         }
-        
+
         RefreshStatusEffects();
     }
-    
+
     private void RefreshStatusEffects() {
         float maxSlowPercentage = 0.0f;
         float totalDamagePerSecond = 0.0f;
-        
+
         foreach (StatusEffect statusEffect in activeStatusEffects) {
             if (statusEffect.slowPercentage > maxSlowPercentage) {
                 maxSlowPercentage = statusEffect.slowPercentage;
             }
-            
+
             totalDamagePerSecond += statusEffect.damagePerSecond;
         }
-        
+
         foreach (TimedStatusEffect timedStatusEffect in timedStatusEffects) {
             if (timedStatusEffect.statusEffect.slowPercentage > maxSlowPercentage) {
                 maxSlowPercentage = timedStatusEffect.statusEffect.slowPercentage;
             }
-            
+
             totalDamagePerSecond += timedStatusEffect.statusEffect.damagePerSecond;
         }
-        
+
         modifiedSpeed = baseSpeed * ((100.0f - maxSlowPercentage) / 100.0f);
         damagePerSecondReceiving = totalDamagePerSecond;
     }
-    
+
     private void UpdateTimedStatusEffects() {
         List<int> toBeDeleted = new List<int>();
-        
+
         for (int i = 0; i < timedStatusEffects.Count; i++) {
```

Comment:
i think a foreach should work here too, and the toBeDeleted list is not needed, just use timedStatusEffects.Remove()
it should find the correct instance as far as I remember